cli/cmd/collector_create.go

```diff
@@ -16,25 +18,28 @@ func (r *runners) InitCollectorCreate(parent *cobra.Command) {
 	cmd.Hidden = true // Not supported in KOTS
 	parent.AddCommand(cmd)
 
-	cmd.Flags().StringVar(&r.args.createCollectorYaml, "yaml", "", "The YAML config for this collector. Use '-' to read from stdin.  Cannot be used with the `yaml-file` flag.")
-	cmd.Flags().StringVar(&r.args.createCollectorYamlFile, "yaml-file", "", "The file name with YAML config for this collector.  Cannot be used with the `yaml` flag.")
+	cmd.Flags().StringVar(&r.args.createCollectorYaml, "yaml", "", "The YAML config for this collector. Use '-' to read from stdin. Cannot be used with the --yaml-file flag.")
+	cmd.Flags().StringVar(&r.args.createCollectorYamlFile, "yaml-file", "", "The file name with YAML config for this collector. Cannot be used with the --yaml flag.")
 	cmd.Flags().StringVar(&r.args.createCollectorName, "name", "", "The name for this collector")
 
 	cmd.RunE = r.collectorCreate
 }
 
 func (r *runners) collectorCreate(cmd *cobra.Command, args []string) error {
-
 	if r.args.createCollectorName == "" {
-		return fmt.Errorf("collector name is required")
+		return errors.New("collector name is required")
 	}
 
 	if r.args.createCollectorYaml == "" && r.args.createCollectorYamlFile == "" {
-		return fmt.Errorf("yaml is required")
+		return errors.New("one of --yaml or --yaml-file is required")
 	}
 
 	if r.args.createCollectorYaml != "" && r.args.createCollectorYamlFile != "" {
-		return fmt.Errorf("only one of yaml or yaml-file may be specified")
+		return errors.New("only one of yaml or yaml-file may be specified")
+	}
+
+	if strings.Contains(r.args.createCollectorYaml, ".yaml") {
```

Comment:
Any string can appear anywhere in the input though.  How about testing if input is a file name like
```
os.Stat(r.args.createCollectorYaml)
```

Reply:
That's definitely an improvement but it'll fail if the filename doesn't exist. So for example, it'll still return `Kubernetes version is required` if the file isn't found. Seems a little confusing to me but maybe that's not a big deal. For now, I modified it to check for the `.yaml` / `.yml` suffix and to check that the input is a single word. I can change it to the `os.Stat` method if you prefer it though.